src/app/loadout-builder/processWorker/processUtils.ts

```diff
@@ -58,3 +81,35 @@ export function canTakeAllSeasonalMods(
   // This will indicate we have iterated over all the mods, it will overshoot the length on success.
   return processedMods.length === modIndex;
 }
+
+/**
+ * See if we can slot all the locked general mods.
+ *
+ * This function need to be kept inline with ../mod-utils#assignAllGeneralMods.
```

Comment:
Can you elaborate a bit on these "must be kept inline with" comments? It seems like they're duplicate versions of the same general functions?

Reply:
Yes they are the same algorithm but they have different outputs/results. 

The one in processUtils returns a boolean to indicate whether the mods can fit and so on, it's basically optimised for performance and doesn't create anything it doesn't need to.

The one in modUtils mutates an object of item hashes to mods, so that we can display which mod is being assigned to which item in the UI. 

I would like to make them the same function and I probably could through passing in the mutated object as an optional parameter. I have only had that thought now so I will see if I can update it and remove the duplication.

Reply:
Oh I also forgot, the types are different too so I would need to do something about that.